Lib/selectors.py

```diff
@@ -338,92 +338,90 @@ def select(self, timeout=None):
         return ready
 
 
-if hasattr(select, 'poll'):
+class _PollLikeSelector(_BaseSelectorImpl):
+    """Base class shared between poll, epoll and devpoll selectors."""
 
-    class PollSelector(_BaseSelectorImpl):
-        """Poll-based selector."""
+    def __init__(self):
+        super().__init__()
+        self._poller = self._poller()
```

Comment:
This looks strange: I assume that self._poller is the poller class (e.g. select, poll, etc).
It's used here but not defined in this base class: maybe make it an abstract property to force its definition?
Also, assigning
self.8poller = self._poller() is very confusing: you're reusing the same member for the instance and the class.
Maybe use poller_cls, or poller_type for the type of the poller, to avoid confusion with the instance?